sprindCrud/src/main/java/com/example/controllers/GamesSpringController.java

```diff
@@ -0,0 +1,114 @@
+
+
+    @RequestMapping(path = "/", method = RequestMethod.GET)
+    public String home(HttpSession session, Model model){
+        String username = (String) session.getAttribute(SESSION_USERNAME);
+        User user = userRepository.findFirstByUsername(username);
+
+
+        if(user != null) {
+            model.addAttribute("user", user);
+
+            List<Game> games = gameRepository.findAll();
+            model.addAttribute("games", games);
+
+            List<Game> usergames = gameRepository.findByUser(user);
+            model.addAttribute("usergames", usergames);
+        }
+        return "home";
+    }
+
+
+    @RequestMapping(path = "/login", method = RequestMethod.POST)
+    public String login(HttpSession session, String username, String password) throws Exception{
+        User user = userRepository.findFirstByUsername(username);
+        if(user == null){
+            user = new User(username, PasswordStorage.createHash(password));
+            userRepository.save(user);
+        }
+        else if(!PasswordStorage.verifyPassword(password, user.getPassword())){
+            throw new Exception("incorrect password");
+        }
+        session.setAttribute(SESSION_USERNAME, username);
+        return "redirect:/";
+    }
+    @RequestMapping(path = "/add-game", method = RequestMethod.POST)
+    public String addgame(HttpSession session, String name, String genre, String platform){
+        String username = (String)session.getAttribute(SESSION_USERNAME);
+        User user = userRepository.findFirstByUsername(username);
+        if(user != null){
+            Game game = new Game(name, genre, platform, user);
+            gameRepository.save(game);
+        }
+        return "redirect:/";
+    }
+
+
+    @RequestMapping(path = "/logout", method = RequestMethod.POST)
+    public String logout(HttpSession session) {
+        session.invalidate();
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/edit", method = RequestMethod.GET)
+    public String edit(HttpSession session,int game, Model model){
+        String username = (String)session.getAttribute(SESSION_USERNAME);
+        User user = userRepository.findFirstByUsername(username);
+        Game gameid = gameRepository.findById(game);
+        if(user != null){
+            model.addAttribute("games", gameid);
+            model.addAttribute("user", user);
+        }
+        return "edit";
+    }
+
+    @RequestMapping(path = "/delete", method = RequestMethod.POST)
+    public String datele(HttpSession session, int id){
+        String username = (String)session.getAttribute(SESSION_USERNAME);
+        User user = userRepository.findFirstByUsername(username);
+        if(user != null){
+            gameRepository.delete(id);
+        }
+        return "redirect:/";
+    }
+
+    @RequestMapping(path = "/update", method = RequestMethod.POST)
+    public String update(HttpSession session,int id, String name, String genre, String platform){
```

Comment:
You didn't have the information before the homework assignment but this could be greatly simplified by putting the `Game` object instead of name, id, genre, and platform